### PR TITLE
KOGITO-8994: drools/optaplanner version update in kie-benchmarks

### PR DESCRIPTION
**Pull request by mbiarnes**
### JIRA
[KOGITO-8994](https://issues.redhat.com/browse/KOGITO-8994)

[kie-benchmarks](https://github.com/kiegroup/kie-benchmarks/pull/231)
[optaplanner](https://github.com/kiegroup/optaplanner/pull/2796)
[drools](https://github.com/kiegroup/drools/pull/5183)

### CODE REVIEW

File: .ci/jenkins/dsl/jobs.groovy [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+* https://github.com/kiegroup/kogito-pipelines/tree/main/dsl/seed/src/main/groovy/org/kie/jenkins/jobdsl.
+*/
+
+import org.kie.jenkins.jobdsl.model.JenkinsFolder
+import org.kie.jenkins.jobdsl.model.JobType
+import org.kie.jenkins.jobdsl.utils.EnvUtils
+import org.kie.jenkins.jobdsl.utils.JobParamsUtils
+import org.kie.jenkins.jobdsl.KogitoJobTemplate
+import org.kie.jenkins.jobdsl.KogitoJobUtils
+import org.kie.jenkins.jobdsl.Utils
```
Comment (radtriste) [inline .ci/jenkins/dsl/jobs.groovy:17]:
```suggestion
import org.kie.jenkins.jobdsl.model.JobType
import org.kie.jenkins.jobdsl.utils.JobParamsUtils
import org.kie.jenkins.jobdsl.KogitoJobTemplate
import org.kie.jenkins.jobdsl.Utils
```

File: .ci/jenkins/dsl/jobs.groovy [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    def jobParams = JobParamsUtils.getBasicJobParams(this, 'kie-benchmarks-update-drools', JobType.TOOLS, "${jenkins_path}/Jenkinsfile.bump-up-version", 'Drools bump up version')
+    JobParamsUtils.setupJobParamsDefaultMavenConfiguration(this, jobParams)
+    jobParams.env.putAll([
+        JENKINS_EMAIL_CREDS_ID: "${JENKINS_EMAIL_CREDS_ID}",
+
+        GIT_BRANCH_NAME: "${GIT_BRANCH}",
+
+        IS_MAIN_BRANCH: "${Utils.isMainBranch(this)}",
+
+        REPO_NAME: 'drools',
```
Comment (radtriste) [inline .ci/jenkins/dsl/jobs.groovy:39]: please use another name. `REPO_NAME` is used to identify the repository against which you run. Here it is `kie-benchmarks`

File: .ci/jenkins/Jenkinsfile.bump-up-version [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    return params.DROOLS_VERSION
+}
+
+String getOptaPlannerVersion() {
+    return params.OPTAPLANNER_VERSION
+}
+
+String getUpdateRepo() {
+    return env.REPO_NAME
+}
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.bump-up-version:77]: Make this more dynamic.
For example, you could have as an environment varialble a `DEPENDENCY_PROJECT`
and then you would have:

```suggestion
String getDroolsVersion() {
    return params.DROOLS_VERSION
}

String getVersion() {
    return params."${DEPENDENCY_PROJECT.toUpperCase()}_VERSION"
}

String getDependencyProject() {
    return env.DEPENDENCY_PROJECT
}
```

File: .ci/jenkins/dsl/jobs.groovy [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+jenkins_path = '.ci/jenkins'
+
+///////////////////////////////////////////////////////////////////////////////////////////
+// kie-benchmarks jobs
+///////////////////////////////////////////////////////////////////////////////////////////
+
+// update drools/optaplanner version in kie-benchmarks
+setupDroolsUpdateJob()
+setupOptaPlannerUpdateJob()
```
Comment (radtriste) [inline .ci/jenkins/dsl/jobs.groovy:27]:
```suggestion
setupUpdateDependencyJob('optaplanner')
setupUpdateDependencyJob('drools')
```

Those 2 methods are linked to one Jenkinsfile. You could make it more generic (goes with the `DEPENDENCY_PROJECT` comment)
and have a single method `setupUpdateDependencyJob`

**Review by radtriste — CHANGES_REQUESTED** [review]

**Review by dupliaka — APPROVED** [review]
LGTM to me

File: .ci/jenkins/Jenkinsfile.bump-up-version (hunk 3 above)
Comment (radtriste) [reply .ci/jenkins/Jenkinsfile.bump-up-version:77]: could be yes. This is just an example. Adapt it as you wish ;)

File: .ci/jenkins/dsl/jobs.groovy [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import org.kie.jenkins.jobdsl.Utils
+
+jenkins_path = '.ci/jenkins'
+
+///////////////////////////////////////////////////////////////////////////////////////////
+// kie-benchmarks jobs
+///////////////////////////////////////////////////////////////////////////////////////////
+
+// update drools/optaplanner version in kie-benchmarks
+// update drools/optaplanner version in kie-benchmarks
```
Comment (radtriste) [inline .ci/jenkins/dsl/jobs.groovy:23]:
```suggestion
```

File: .ci/jenkins/dsl/jobs.groovy [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    def jobParams = JobParamsUtils.getBasicJobParams(this, "kie-benchmarks-update-${updateRepoName}", JobType.TOOLS, "${jenkins_path}/Jenkinsfile.bump-up-version", "${updateRepoName} bump up version")
+    JobParamsUtils.setupJobParamsDefaultMavenConfiguration(this, jobParams)
+    jobParams.env.putAll([
+        JENKINS_EMAIL_CREDS_ID: "${JENKINS_EMAIL_CREDS_ID}",
+
+        GIT_BRANCH_NAME: "${GIT_BRANCH}",
+
+        IS_MAIN_BRANCH: "${Utils.isMainBranch(this)}",
+
+        UPDATE_REPO_NAME: "${updateRepoName}",
```
Comment (radtriste) [inline .ci/jenkins/dsl/jobs.groovy:37]: unsused. can be removed

File: .ci/jenkins/Jenkinsfile.bump-up-version [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    }
+    post{
+        always{
+            cleanWs()
+        }
+    }
+}
+
+String getVersion() {
+    return params."${getDependencyProject().toUpperCase()}_VERSION"
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.bump-up-version:63]: I just think that you don't even need the dependency name for the parameter.
The parameter could be just named something like `NEW_VERSION`

File: .ci/jenkins/Jenkinsfile.bump-up-version [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+                        checkoutRepo("${repoName}", "${getTargetBranch()}")
+                    }
+                }
+            }
+        }
+        stage('bump up version') {
+            steps {
+                script {
+                    dir("${repoName}") {
+                        String commitMsg = "bumped up ${getDependencyProject()} version to next SNAPSHOT"
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.bump-up-version:42]:
```suggestion
                        String commitMsg = "bumped up ${getDependencyProject()} version to ${getVersion()}"
```

**Review by radtriste — APPROVED** [review]
minor comments

File: .ci/jenkins/Jenkinsfile.bump-up-version [hunk 9]
```diff
@@ -0,0 +1,101 @@
+import org.jenkinsci.plugins.workflow.libs.Library
+@Library('jenkins-pipeline-shared-libraries')_
+
+import org.kie.jenkins.MavenCommand
+
+def repoName = 'kie-benchmarks'
+
+pipeline {
+    agent {
+        label 'kie-rhel8 && !built-in'
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.bump-up-version:10]: Thinking that it should be enough.
`kie-rhel8` is a 16G mem node, and we don't need it here IMO
```suggestion
        label 'rhel8 && !built-in'
```

File: .ci/jenkins/Jenkinsfile.bump-up-version [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    }
+
+    stages {
+        stage('Initialize') {
+            steps {
+                script {
+                    cleanWs()
+
+                    dir("${repoName}") {
+                        checkoutRepo("${repoName}", "${getTargetBranch()}")
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.bump-up-version:33]:
```suggestion
                        checkoutRepo(repoName, getTargetBranch())
```

File: .ci/jenkins/Jenkinsfile.bump-up-version [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+        timeout(time: 30, unit: 'MINUTES')
+    }
+
+    stages {
+        stage('Initialize') {
+            steps {
+                script {
+                    cleanWs()
+
+                    dir("${repoName}") {
```
Comment (radtriste) [inline .ci/jenkins/Jenkinsfile.bump-up-version:32]:
```suggestion
                    dir("${repoName}") {
                        deleteDir()
```

**Review by radtriste — APPROVED** [review]

**Pull request merged**